plasma_framework/docs/design/exit_bond_and_bounty_mechanics.md

```diff
@@ -0,0 +1,190 @@
+The user who processes an exit, gets the bounty reward for processing an exit. The cost of processing multiple exits from the exit queue at once could be of less cost than processing all those exits individually. However, the bounty amount associated with each exit is given out to the processor.
+
+**_On processing exit:_**
+*Bond - Bounty -> from `Exit Game Contract` to `Exit owner`
+Bounty -> from `Exit Game Contract` to `Exit processor`*
+
+### For In-Flght Exits
+
+The Bond/Bounty collection for in-flight exits is similar. However, for an in-flight exit, the Exit Bounty is a poriton of the Piggyback Bond. Since only the piggybacked input/output from an in-flight exit can be exited, it only makes sense to have the bounty for each piggyback from the respective owner. The size of the Piggyback Bond can be obtained by calling `piggybackBondSize()`.
+
+**_After Piggybacking an input/output:_**
+*Piggyback Bond -> from `I/O Owner` to `Exit Game Contract`*
+
+For any Input/Output spent challenge to the exit, the whole piggyback bond is transferred to the challenger. 
+
+**_On succesfull challenge:_**
+*Piggyback Bonds -> from `Exit Game Contract` to `Challenger`*
+
+Both the standard and in-flight exit land on the same exit queue and the process exit bounty associated with each exit (or specifically each input/output) is awarded to the processor.
+
+**_On processing exit:_**
+*Bond - Bounty -> from `Exit Game Contract` to `I/O Owner`
+Bounty -> from `Exit Game Contract` to `Exit processor`*
+
+## Bond and Bounty Adjustment Mechanism
+
+
+### The Updatable Structure
+
+The Exit Bond follows an updatable pattern which allows it to be updated (by the maintainer) and reflect within a time period (2 days from the last update). The size of the Exit Bounty can also be updated as long as it is lower (or equal) than the Exit Bond Size and is within the upper-lower bounds. 
+
+```
+struct Params {
+        uint128 previousBondSize;
+        uint128 updatedBondSize;
+        uint128 previousExitBountySize;
+        uint128 updatedExitBountySize;
+        uint128 effectiveUpdateTime;
+        uint16 lowerBoundDivisor;
+        uint16 upperBoundMultiplier;
+    }
+
+```
+
+`Params` is a structure for storing the bond size:
+
+1. `previousBondSize` reflects the bond size, if `now` < `effectiveUpdateTime`.
+2. `updatedBondSize` field reflects the bond size, when when the effectiveUpdateTime period has passed and `now` > `effectiveUpdateTime`.
+3. `previousExitBountySize` reflects the bounty size, if `now` < `effectiveUpdateTime`.
+4. The `updatedExitBountySize` field reflects the bounty size, when the effectiveUpdateTime period has passed and `now` > `effectiveUpdateTime`.
+5. The `effectiveUpdateTime` field denotes the timestamp at which the updated bond and bounty size should reflect and be put to use. The value of the timestamp is 2 days from the last update timestamp.
+6. `lowerBoundDivisor` is a factor which limits the bond to be updated to a certain minimum
```

Comment:
Does the Exit Bounty need an upper-bound and a lower-bound? or a free moving value within the Bond size should be accepted

Reply:
I think any value from 0 to the bond size is fine.
We don't need to force it to e.g.  a certain percentage of the bond